app/models/manageiq/providers/amazon/inventory/ems_event_collection.rb

```diff
@@ -0,0 +1,25 @@
+class ManageIQ::Providers::Amazon::Inventory::EmsEventCollection
+  attr_accessor :targets
+
+  def initialize(targets)
+    @targets = targets
+  end
+
+  def all_related_ems_events(ems)
+    # We want all EmsEvents around the time of collected EmsEvents, and to the future since AWS send all related events
+    # at once
+    EmsEvent.where(:ems_id => ems.id).where(:timestamp => datetime_min..DateTime::Infinity.new).order("timestamp DESC")
+  end
+
+  def datetime_min
+    targets.collect(&:timestamp).min - 3.minutes
```

Comment:
it's a first draft condition to find related events, usually the similar events should have a CloudTrail id, but it's not true for some reason, when deploying through stack.